### PR TITLE
Added noise parameter for http payload conditional fuzzing support

### LINKED ISSUE #3005: Add noise level to configure fuzzing payload count at runtime based on user preference
### Please describe your feature request:
Currently there's no way to specify how much noise we want to generate during fuzzing. As an example - https://github.com/projectdiscovery/fuzzing-templates/blob/main/crlf/crlf-injection.yaml - this template sends 30 payloads by default.

Somehow we should make it configurable and use a low count of payloads by default. Usually this is done by having a command line flag like `-noise <low/medium/high>` and using the provided value to configure payload sets.

A potential example of how this would look like - 

```yaml
payloads:
  escape:
     low:
        - "%00"
        - "%0a"
        - "%0a%20"
        - "%0d"
     medium:
        - "%0d%09"
        - "%0d%0a"
        - "%0d%0a%09"
        - "%0d%0a%20"
        - "%0d%20"
        - "%20"
        - "%20%0a"
        - "%20%0d"
        - "%20%0d%0a"
        - "%23%0a"
        - "%23%0a%20"
        - "%23%0d"
        - "%23%0d%0a"
        - "%23%oa"
        - "%25%30"
        - "%25%30%61"
      high:
        - "%2e%2e%2f%0d%0a"
        - "%2f%2e%2e%0d%0a"
        - "%2f..%0d%0a"
        - "%3f"
        - "%3f%0a"
        - "%3f%0d"
        - "%3f%0d%0a"
        - "%e5%98%8a%e5%98%8d"
        - "%e5%98%8a%e5%98%8d%0a"
        - "%e5%98%8a%e5%98%8d%0d"
        - "%e5%98%8a%e5%98%8d%0d%0a"
        - "%e5%98%8a%e5%98%8d%e5%98%8a%e5%98%8d"
        - "%u0000"
        - "%u000a"
        - "%u000d"
        - "\r"
        - "\r%20"
        - "\r\n"
        - "\r\n%20"
        - "\r\n\t"
        - "\r\t"
```

Other suggestions on how to achieve this are welcomed.

### Describe the use case of this feature:

Allow user configurable fuzzing scans with configurable aggression.

### PR DESCRIPTION
**Pull request by Ice3man543**
## Proposed changes

Closes #3005

```yaml
id: crlf-injection

info:
  name: CRLF Injection
  author: pdteam
  severity: low
  tags: crlf,dast

requests:
  - method: GET
    path:
      - "{{BaseURL}}"

    payloads:
      escape:
        low: 
          - "%00"
          - "%0a"
          - "%0a%20"
          - "%0d"
        medium:
          - "%0d%09"
          - "%0d%0a"
          - "%0d%0a%09"
          - "%0d%0a%20"
          - "%0d%20"
          - "%20"
          - "%20%0a"
          - "%20%0d"
          - "%20%0d%0a"
          - "%23%0a"
          - "%23%0a%20"
          - "%23%0d"
          - "%23%0d%0a"
          - "%23%oa"
          - "%25%30"
        high:
          - "%25%30%61"
          - "%2e%2e%2f%0d%0a"
          - "%2f%2e%2e%0d%0a"
          - "%2f..%0d%0a"
          - "%3f"
          - "%3f%0a"
          - "%3f%0d"
          - "%3f%0d%0a"
          - "%e5%98%8a%e5%98%8d"
          - "%e5%98%8a%e5%98%8d%0a"
          - "%e5%98%8a%e5%98%8d%0d"
          - "%e5%98%8a%e5%98%8d%0d%0a"
          - "%e5%98%8a%e5%98%8d%e5%98%8a%e5%98%8d"
          - "%u0000"
          - "%u000a"
          - "%u000d"
          - "\r"
          - "\r%20"
          - "\r\n"
          - "\r\n%20"
          - "\r\n\t"
          - "\r\t"

    fuzzing:
      - part: query
        type: postfix
        fuzz:
          - "{{escape}}Set-Cookie:crlfinjection=crlfinjection"

    stop-at-first-match: true
    matchers:
      - type: regex
        part: header
        regex:
          - '(?m)^(?:Set-Cookie\s*?:(?:\s*?|.*?;\s*?))(crlfinjection=crlfinjection)(?:\s*?)(?:$|;)'
```

Example run

```bash
./nuclei -t template.yaml -u "https://example.com/?test=value" -v

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   v2.8.4-dev

                projectdiscovery.io

[INF] Using Nuclei Engine 2.8.4-dev (development)
[INF] Using Nuclei Templates 9.3.3 (latest)
[INF] Templates added in last update: 238
[INF] Templates loaded for scan: 1
[INF] Targets loaded for scan: 1
[VER] [crlf-injection] Sent HTTP request to https://example.com/?test=value%2500Set-Cookie%3Acrlfinjection%3Dcrlfinjection
[VER] [crlf-injection] Sent HTTP request to https://example.com/?test=value%250aSet-Cookie%3Acrlfinjection%3Dcrlfinjection
[VER] [crlf-injection] Sent HTTP request to https://example.com/?test=value%250a%2520Set-Cookie%3Acrlfinjection%3Dcrlfinjection
[VER] [crlf-injection] Sent HTTP request to https://example.com/?test=value%250dSet-Cookie%3Acrlfinjection%3Dcrlfinjection
[INF] No results found. Better luck next time!
 
 ./nuclei -t template.yaml -u "https://example.com/?test=value" -v -noise medium

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   v2.8.4-dev

                projectdiscovery.io

[INF] Using Nuclei Engine 2.8.4-dev (development)
[INF] Using Nuclei Templates 9.3.3 (latest)
[INF] Templates added in last update: 238
[INF] Templates loaded for scan: 1
[INF] Targets loaded for scan: 1
[VER] [crlf-injection] Sent HTTP request to https://example.com/?test=value%2500Set-Cookie%3Acrlfinjection%3Dcrlfinjection
[VER] [crlf-injection] Sent HTTP request to https://example.com/?test=value%250aSet-Cookie%3Acrlfinjection%3Dcrlfinjection
[VER] [crlf-injection] Sent HTTP request to https://example.com/?test=value%250a%2520Set-Cookie%3Acrlfinjection%3Dcrlfinjection
[VER] [crlf-injection] Sent HTTP request to https://example.com/?test=value%250dSet-Cookie%3Acrlfinjection%3Dcrlfinjection
[VER] [crlf-injection] Sent HTTP request to https://example.com/?test=value%250d%2509Set-Cookie%3Acrlfinjection%3Dcrlfinjection
[VER] [crlf-injection] Sent HTTP request to https://example.com/?test=value%250d%250aSet-Cookie%3Acrlfinjection%3Dcrlfinjection
...
```

## Checklist

- [x] Pull request is created against the [dev](https://github.com/projectdiscovery/nuclei/tree/dev) branch
- [ ] All checks passed (lint, unit/integration/regression tests etc.) with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by Mzack9999** [comment]
As the change is quite structural (YAML syntax), what do you think about creating different templates with different noise tags and reusing all common parts with https://github.com/projectdiscovery/nuclei/pull/1767?

**Comment by Ice3man543** [comment]
@Mzack9999 that would make it quite complicated and probably unusable. This was the cleanest solution i could think of while having all types of payload sets in a single file.

**Review by Mzack9999 — APPROVED** [review]
lgtm implementation - I'd recommend double checking with @ehsandeep the yaml syntax proposed change

File: v2/cmd/nuclei/main.go [hunk 1]
```diff
@@ -195,6 +195,7 @@ on extensive configurability, massive extensibility and ease of use.`)
 		flagSet.BoolVarP(&options.ForceAttemptHTTP2, "force-http2", "fh2", false, "force http2 connection on requests"),
 		flagSet.BoolVarP(&options.EnvironmentVariables, "env-vars", "ev", false, "enable environment variables to be used in template"),
 		flagSet.StringVarP(&options.ClientCertFile, "client-cert", "cc", "", "client certificate file (PEM-encoded) used for authenticating against scanned hosts"),
+		flagSet.StringVarP(&options.Noise, "noise", "ne", "low", "noise level for http fuzzing (accepted: low,medium,high)"),
```
Comment (tarunKoyalwar) [inline v2/cmd/nuclei/main.go:198]: @Ice3man543 , what do you think of shortflag `nl` ?

**Review by tarunKoyalwar — APPROVED** [review]
lgtm!

**Pull request closed without merging**